eclipse/rdf4j
# GH-2145: named graph support in federation engine

GH-2145: fix context retrieval in federation engine

* context IDs retrieved from the endpoints have to be materialized as
they are only available while the connection is open
* add unit test infrastructure for named graph tests and define some
test data


GH-2145: add named graph handling for #getStatements API

Add named graph support to the RepositoryConnection API using the
getStatements method. Also covered with various unit tests.

Note that this change provides already the infrastructure to further add
support for named graph queries.


GH-2145: add named graph handling for simple federated SELECT patterns

* support GRAPH clause around single basic graph pattern
* support FROM clause in queries


* add named graph handling for simple federated SELECT patterns
** support GRAPH clause around single basic graph pattern
** support FROM clause in queries

* fix handling of base URIs for query parsing in FedX

* add named graph support for bound joins

GH-2149 GH-2145 use new compliance test suite for testing of FedX

GitHub issue resolved: #2145


---- 
PR Author Checklist: 

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [ ] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.
